salt/modules/xbps-pkg.py

```diff
@@ -0,0 +1,641 @@
+    for line in out.splitlines():
+        if not line:
+            continue
+        try:
+            pkg, ver = line.split()[0].rsplit('-', 1)
+        except (ValueError, IndexError):
+            log.error('xbps-query: Unexpected formatting in '
+                      'line: "{0}"'.format(line))
+            continue
+
+        log.trace('pkg={0} version={1}'.format(pkg, ver))
+        ret[pkg] = ver
+
+    return ret
+
+
+def latest_version(*names, **kwargs):
+    '''
+    Return the latest version of the named package available for upgrade or
+    installation. If more than one package name is specified, a dict of
+    name/version pairs is returned.
+
+    If the latest version of a given package is already installed, an empty
+    string will be returned for that package.
+
+    .. versionadded:: XXX 201X.XX
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' pkg.latest_version <package name>
+        salt '*' pkg.latest_version <package1> <package2> <package3> ...
+    '''
+
+    # Why using 'xbps-install -un' and not 'xbps-query -R':
+    # if several repos, xbps-query will produces this kind of output,
+    # that is difficult to handle correctly:
+    #     [*] salt-2015.8.3_2 Remote execution system ...
+    #     [-] salt-2015.8.3_1 Remote execution system ...
+    #
+    # XXX 'xbps-install -un pkg1 pkg2' won't produce any info on updatable pkg1 
+    #     if pkg2 is up-to-date. Bug of xbps 0.51, probably get fixed in 0.52.
+    #     See related issue https://github.com/voidlinux/xbps/issues/145
+    #
+    # sample outputs of 'xbps-install -un':
+    #     fuse-2.9.4_4 update i686 http://repo.voidlinux.eu/current 298133 91688
+    #     xtools-0.34_1 update noarch http://repo.voidlinux.eu/current 21424 10752
+    #     Package 'vim' is up to date.
+
+    refresh = salt.utils.is_true(kwargs.pop('refresh', True))
+    
+    if len(names) == 0:
+        return ''
+
+    # Refresh repo index before checking for latest version available
+    if refresh:
+        refresh_db()
+
+    # Initialize the dict with empty strings
+    ret = {}
+    for name in names:
+        ret[name] = ''
+
+    # retrieve list of updatable packages
+    # ignore return code since 'is up to date' case produces retcode==17 (xbps 0.51)
+    cmd = '{0} {1}'.format('xbps-install -un', ' '.join(names))
+    out = __salt__['cmd.run'](cmd, ignore_retcode=True, 
+                                   output_loglevel='trace')
+    for line in out.splitlines():
+        if not line:
+            continue
+        if line.find(' is up to date.') != -1:
+            continue
+        # retrieve tuple pkgname version
+        try:
+            pkg, ver = line.split()[0].rsplit('-', 1)
+        except (ValueError, IndexError):
+            log.error('xbps-query: Unexpected formatting in '
+                      'line: "{0}"'.format(line))
+            continue
+
+        log.trace('pkg={0} version={1}'.format(pkg, ver))
+        if pkg in names:
```

Comment:
Confirmed again that this issue is a non-issue since `continue` has only the normal special meaning in `try: except:` blocks.